internal/bowtie/test/dns_block_list_test.go

```diff
@@ -0,0 +1,77 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/bowtieworks/terraform-provider-bowtie/internal/bowtie/provider"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+const (
+	resourceType = "bowtie_dns_block_list"
+	resourceName = "test"
+
+	blName       = "Test DNS Block List"
+	blNameChange = "Different DNS Block List name"
+	blUrl        = "https://gist.githubusercontent.com/tylerjl/a98e82a7c62207dcd91aad47110e135d/raw/409e482fa067f0ca21c62f916a2bfb8f8b83bcc4/block.txt"
+	blUrlChange  = "https://gist.githubusercontent.com/tylerjl/a98e82a7c62207dcd91aad47110e135d/raw/c128c2d15ddaf787eed4efa61960f984ca61995a/block.txt"
+	blOverride   = `ipchicken.com
+downloadmoreram.com
+`
+	blOverrideChange = `ipchicken.com
+downloadmoreram.com
+neopets.com
+`
+)
+
+func TestDNSBlockListResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: provider.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Basic tests for upstream URLs
+			{
+				Config: provider.ProviderConfig + fmt.Sprintf(`
```

Comment:
FWIW I like the tmpl pattern I used on mine makes the copy paste a little less egegious in my opinion but also not critical